src/bgen/Attributes.cs

```diff
@@ -125,6 +125,11 @@ public PostGetAttribute (string name)
 	public string MethodName { get; set; }
 }
 
+[AttributeUsage (AttributeTargets.Class | System.AttributeTargets.Enum, AllowMultiple = false)]
+public class BindingTypeAttribute : Attribute {
+	public string Name { get; set; }
+}
+
```

Comment:
This won't work: with rgen, all the attributes we need we'll have be to included in our product assembly, we can't use a separate attribute assembly (because then our product assembly would end up with a reference to the attribute assembly, which complicates everything _a lot_).

Reply:
Yeah, I thought that too. We probably have to add it in the fina assembly but the roslyn analyser cannot inject it. So we would force people to reference our product assembly to create bindings. Any objections about that? 

We have think about a namespace.

Reply:
> So we would force people to reference our product assembly to create bindings. Any objections about that?

Nope, imho it doesn't make much sense to not reference our product assembly (where do you get the `NSObject` type from otherwise?)